genai-perf/tests/test_triton_generate_converter.py

```diff
@@ -0,0 +1,189 @@
+
+    def test_convert_with_extra_inputs(self):
+        generic_dataset = self.create_generic_dataset(
+            [DataRow(texts=["extra_input_prompt"])]
+        )
+
+        extra_inputs = {"temperature": 0.7, "top_p": 0.9}
+        config = InputsConfig(
+            extra_inputs=extra_inputs,
+            tokenizer=get_empty_tokenizer(),
+        )
+
+        converter = TritonGenerateConverter()
+        result = converter.convert(generic_dataset, config)
+
+        expected_result = {
+            "data": [
+                {
+                    "payload": [
+                        {
+                            "text_input": ["extra_input_prompt"],
+                            "temperature": 0.7,
+                            "top_p": 0.9,
+                        }
+                    ]
+                }
+            ]
+        }
+
+        assert result == expected_result
+
+    def test_convert_with_streaming(self):
+        generic_dataset = self.create_generic_dataset(
+            [DataRow(texts=["streaming_prompt"])]
+        )
+
+        config = InputsConfig(
+            add_stream=True,
+            tokenizer=get_empty_tokenizer(),
+        )
+
+        converter = TritonGenerateConverter()
+        result = converter.convert(generic_dataset, config)
+
+        expected_result = {
+            "data": [
+                {
+                    "payload": [
+                        {
+                            "text_input": ["streaming_prompt"],
+                            "stream": True,
+                        }
+                    ]
+                }
+            ]
+        }
+
+        assert result == expected_result
+
+    def test_convert_with_output_tokens_mean(self, monkeypatch):
+        generic_dataset = self.create_generic_dataset(
+            [DataRow(texts=["tokens_mean_prompt"])]
+        )
+
+        config = InputsConfig(
+            output_tokens_mean=100,
+            output_tokens_stddev=10,
+            tokenizer=get_empty_tokenizer(),
+        )
+
+        def mock_sample_bounded_normal(mean, stddev, lower):
+            assert mean == 100
+            assert stddev == 10
+            assert lower == 1
+            return 95
+
+        monkeypatch.setattr(
+            "genai_perf.inputs.converters.triton_generate_converter.sample_bounded_normal",
+            mock_sample_bounded_normal,
+        )
+
+        converter = TritonGenerateConverter()
+        result = converter.convert(generic_dataset, config)
+
+        expected_result = {
+            "data": [
+                {
+                    "payload": [
+                        {
+                            "text_input": ["tokens_mean_prompt"],
+                            "max_tokens": 95,
+                        }
+                    ]
+                }
+            ]
+        }
+
+        assert result == expected_result
+
+    def test_convert_empty_dataset(self):
+        generic_dataset = GenericDataset(files_data={})
```

Comment:
Is empty dataset possible with our current GAP? If so, should we let it pass without any warning/error?

Reply:
Theoretically, yes. The retrievers decide which data to provide, the converters turn it into the right format. If a retriever  provides no data, it'd be good if the converter doesn't fail spectacularly. I don't know if it's possible with our current retrievers (maybe if the input file is empty), but it could definitely be possible when we add a new retriever.

I was divided on whether to include this test. I think it's very optional, but we have it for a few other converters. We could always add a check if an empty dataset was retrieved before we call the converter to return an exception, but I don't think it's necessary now. If we want to do so, I'd prefer to push it into its own ticket/PR.

Would you prefer for this test to be removed?

Reply:
Sounds good, removed it!